sympy/matrices/expressions/special.py

```diff
@@ -290,10 +294,66 @@ def _eval_inverse(self):
             return Inverse(self)
 
     def _eval_as_real_imag(self):
-        return (self, ZeroMatrix(*self.shape))
+        return self, ZeroMatrix(*self.shape)
 
     def _eval_conjugate(self):
         return self
 
     def _entry(self, i, j, **kwargs):
         return S.One
+
+
+class MatrixUnit(MatrixExpr):
+    """
+    Matrix with only one nonzero entry with value 1.
+
+    https://en.wikipedia.org/wiki/Matrix_unit
+    """
+    def __new__(cls, i, j, shape):
```

Comment:
I'm not sure if these are the best parameters
because other matrix symbols are used to take rows and cols as separate parameters than shape.